tardis/visualization/tools/sdec_plot.py

```diff
@@ -1013,31 +1223,75 @@ def _plot_emission_mpl(self):
                 label="Other elements",
             )
 
-        elements_z = self.elements
         # Contribution from each element
-        for i, atomic_number in enumerate(elements_z):
-            # Add a try catch because elemets_z comes from the total contribution of absorption and emission.
-            # Therefore it's possible that something in elements_z is not in the emission df
+        # Create new variables to keep track of the last atomic number that was plotted
+        # This is used when plotting species incase an element was given in the list
+        # This is to ensure that all ions of that element are grouped together
+        # ii is to track the colour index
+        # e.g. if Si is given in species_list, this is to ensure Si I, Si II, etc. all have the same colour
+        ii = 0
+        previous_atomic_number = 0
+        for i, identifier in enumerate(self.species):
+            if self._species_list is not None:
+                # Get the ion number and atomic number for each species
+                ion_number = identifier % 100
+                atomic_number = (identifier - ion_number) / 100
+                if previous_atomic_number == 0:
+                    # If this is the first species being plotted, then take note of the atomic number
+                    # don't update the colour index
+                    ii = ii
+                    previous_atomic_number = atomic_number
+                elif previous_atomic_number in self._keep_colour:
+                    # If the atomic number is in the list of elements that should all be plotted in the same colour
+                    # then dont update the colour index if this element has been plotted already
+                    if previous_atomic_number == atomic_number:
+                        ii = ii
+                        previous_atomic_number = atomic_number
+                    else:
+                        # Otherwise, increase the colour counter by one, because this is a new element
+                        ii = ii + 1
+                        previous_atomic_number = atomic_number
+                else:
+                    # If this is just a normal species that was requested then increment the colour index
+                    ii = ii + 1
+                    previous_atomic_number = atomic_number
+                # Calculate the colour of this species
+                color = self.cmap(ii / len(self._species_name))
+
+            else:
+                # If you're not using species list then this is just a fraction based on the total
+                # number of columns in the dataframe
+                color = self.cmap(i / len(self.species))
+            # Add a try catch because the identifier comes from the total contribution of absorption and emission.
+            # Therefore it's possible that something in list is not in the emission df
```

Comment:
It's possible that the color preparation can be done separately from the plotting.

Reply:
This is implemented now.